codegen/module.go

```diff
@@ -1222,6 +1260,9 @@ type ModuleClass struct {
 	DependedBy  []string
 	types       map[string]BuildGenerator
 
+	// Every module instance of this module class imports all instances of a module class that is implicitly depended on
+	ImplicitDependsOn []string
```

Comment:
This approach is interesting, I can see it would make sense for certain module class types, however I don't think it is the way to solve what we want to achieve with this PR. `adapter` and `middleware` are fundamentally the same abstraction, the fact that adapters are mandatory for every endpoint does not add more meaning to the middleware abstraction (at least not enough for a spin off), what it introduces is more a rule for a subset of middlewares. 

If the goal here is no more than "to have a list of default middlewares that every endpoint implicitly depends on", then I think it is an implementation detail of the endpoint generation and needs not level up to an approach that alters high level abstractions (which should be kept simple). I am a fan of explicit, but I admit having each endpoint explicitly declare dependency on the set of default middlewares hurts user experience and sometimes is error prone. Maintaining a similar file as you did with https://github.com/uber/zanzibar/pull/552/files#diff-223333e63915149a0b7e1a4ad57fc256 in middelwares directory(I can also see the file in endpoints dir making sense in a way) is probably a reasonable trade off.